### ISSUE TITLE
Android partial wakelock for 75%+ time drains batter

### ISSUE THREAD

**Issue by grassick** [mauron85/cordova-plugin-background-geolocation#417]
## Your Environment
* Plugin version: 3.0.0-alpha.24
* Platform: Android
* OS version: 5.0.1
* Device manufacturer and model: Galaxy S4
* Cordova version (`cordova -v`): 8.0.0
* Cordova platform version (`cordova platform ls`): Android 6.4.0
* Plugin configuration options: None
* Link to your project:

## Context

Both distance and activity provider drain battery much more quickly than expected (12+%/hr). Using BetterBatteryStats and other tools show that the phone is almost continually awake when BackgroundGeolocation is running.

## Expected Behavior
I expected the phone to be asleep mostly between location detections.

## Actual Behavior
Phone was awake 77% of time due to plugin.
![image](https://user-images.githubusercontent.com/1238335/39483420-4fd6d604-4d40-11e8-9e4e-bd81f0dd1bb8.png)

## Possible Fix

Only wake up every N minutes to get a location?

## Steps to Reproduce
1. Create an app using the plugin and configure as follows:
```
    BackgroundGeolocation.configure({
      locationProvider: BackgroundGeolocation.ACTIVITY_PROVIDER,
      desiredAccuracy: BackgroundGeolocation.MEDIUM_ACCURACY,
      debug: false,
      notificationTitle: 'Surveyor Location Tracking',
      notificationText: 'Enabled',
      interval: 1000 * 5 * 60,
      fastestInterval: 1000 * 5 * 60,
      activitiesInterval: 60000
    })

2. Start app and close phone
3. Observe that phone spends time awake, not sleeping


## Context
It prevents this from being used in the field, as an awake phone would drain the batteries too fast.

## Debug logs

I could create as needed.

**Comment by grassick** [comment]
I also tried the distance filter but that was even worse in terms of wakelocks. It showed the phone being continually awake.

**Comment by mauron85** [comment]
Yeah I agree, it's time to investigate if those CPU wakeLocks are still necessary.

**Comment by mauron85** [comment]
Relevant discussion from react-native variant https://github.com/mauron85/cordova-plugin-background-geolocation/issues/417

**Comment by grassick** [comment]
The link is just to this issue... can you double-check it?

**Comment by mauron85** [comment]
Oh you are right. ;) Should be this one. https://github.com/mauron85/react-native-background-geolocation/issues/180

**Comment by grassick** [comment]
I removed the wakelock from the activity provider and it works like a charm in Android. 5% awake time! 

Now, my worry is that we don't keep a wakelock when firing the Javascript `location` event. It seems to me that it would be better to pass a second parameter to the event callback which is a function that must to be called to release the wakelock. That way we'd be guaranteed that the JS code would stay awake long enough to handle the location event properly. What do you think? I have no idea how the iOS side works, unfortunately.

**Comment by mauron85** [comment]
>That way we'd be guaranteed that the JS code would stay awake long enough to handle the location event properly.

I would say, this only affects Android, as no wakeLocks are used in iOS. I know that cordova keeps js timers running in background, so it's possible that we don't need any wakelocks at all.  I don't have any actual recommendation. Maybe bit of testing. Keep in mind that plugin core is shared with react-native too.

**Comment by grassick** [comment]
I've removed the wakelocks in both providers and it works fine for me in Android. It's still calling the Javascript callback and even allowing an AJAX post from within the callback. Did you want a PR?

**Comment by mauron85** [comment]
@grassick Yes, you're very welcome to do so ;)
